native-image-config/build.gradle.kts

```
@@ -26,6 +26,18 @@ plugins {
     `jvm-toolchains`
 }
 
+// If `-Pscratch` is specified, do not source the previously generated config at core/src/main/resources/META-INF/native-image
+// otherwise, the previously generated config will be merged into the newly generated config.
+val shouldGenerateFromScratch = project.findProperty("scratch").let {
```

Question) Would `hasProperty` be more simple (and require less explanation)?

```suggestion
val shouldGenerateFromScratch = project.hasProperty("scratch").let {
```

---

I wanted to make sure our build explicitly rejects the flags like `-Pscratch=false`, which would otherwise have been treated as `true`.